plugins/filter_modify/modify.c

```diff
@@ -69,20 +111,15 @@ static void teardown(struct filter_modify_ctx *ctx)
 
     mk_list_foreach_safe(head, tmp, &ctx->conditions) {
         condition = mk_list_entry(head, struct modify_condition, _head);
-        mk_list_del(&condition->_head);
+        if (!mk_list_entry_is_orphan(&condition->_head)) {
```

Comment:
I prefer to include the list deletion in `rule_free` since the list is a member of struct.

Reply:
I don't understand this answer, following that logic it would make sense to move the removal of the condition instance from the list to `condition_free` to match the logic applied for the `modify_rule` instances. 

Could you please change that?

Reply:
I moved it.
https://github.com/fluent/fluent-bit/pull/7378/commits/a666ae3115f88d2f9230959bf594fb43a0cfeee7

Reply:
Thank you very much @nokute78, I really appreciate it.